### PR TITLE
Scalatest

### PR DESCRIPTION
**Pull request by denen99**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -6,7 +6,8 @@ lazy val root = (project in file(".")).
     retrieveManaged := true,
     libraryDependencies ++= Seq(
       "com.typesafe" % "config" % "1.3.0",
-      "org.specs2" %% "specs2-core" % "3.7" % "test",
+      "org.scalatest" % "scalatest_2.11" % "3.0.1" % "test",
```
Comment (joprice) [inline build.sbt]: This can be `"org.scalatest" %% "scalatest" % "3.0.1" % Test`

File: build.sbt [hunk 2]
```diff
@@ -25,6 +26,10 @@ lazy val root = (project in file(".")).
     )
 )
 
+fork in Test := true // allow to apply extra setting to Test
```
Comment (joprice) [inline build.sbt]: Took me a second to get this. Maybe 'forking is required in order to set `javaOptions in Test`'?

Comment (denen99) [inline build.sbt]: Seems so.. Yes

File: build.sbt (hunk 1 above)
Comment (denen99) [inline build.sbt]: will update

File: src/main/scala/org/dberg/hubot/robot/Robot.scala [hunk 3]
```diff
@@ -12,18 +12,21 @@ trait RobotComponent {
   def robotService: RobotService
   def adapter: BaseAdapter
   def listeners: Seq[Listener]
-  def middleware: Seq[Middleware]
+  def middleware: List[Middleware]
 
   class RobotService extends StrictLogging {
 
     private def processMiddlewareRec(
```
Comment (joprice) [inline src/main/scala/org/dberg/hubot/robot/Robot.scala]: Did you try adding @tailrec to this?  Looks like it only calls itself.

File: src/test/scala/org/dberg/hubot/BrainTestSuite.scala [hunk 4]
```diff
@@ -0,0 +1,34 @@
+package org.dberg.hubot
+
+import org.scalatest.DoNotDiscover
+import scodec.codecs.ImplicitCodecs
+
+@DoNotDiscover
```
Comment (joprice) [inline src/test/scala/org/dberg/hubot/BrainTestSuite.scala]: I've never seen this one. What's DoNotDiscover for?

**Comment by denen99** [comment]
@joprice can you have a look and let me know if you see any other fixes needed ?  tests are passing and the new return types make it much easier

**Comment by joprice** [comment]
Looks good!

**Pull request merged**